rust/gui-client/src-tauri/src/client/logging.rs

```diff
@@ -123,23 +124,65 @@ pub(crate) async fn clear_logs_inner() -> Result<()> {
                 return Err(error.into());
             }
         };
+        let mut paths = vec![];
         while let Some(entry) = dir.next_entry().await? {
-            let path = entry.path();
-            if let Err(error) = tokio::fs::remove_file(&path).await {
-                tracing::error!(
-                    ?error,
-                    path = path.display().to_string(),
-                    "Error while removing log file"
-                );
-                // We'll return the most recent error, it loses some information but it's better than nothing.
-                result = Err(error);
+            paths.push(entry.path());
+        }
+
+        let to_delete = choose_logs_to_delete(&paths);
+        for path in &to_delete {
+            if let Err(e) = tokio::fs::remove_file(path).await {
+                result = Err(e);
             }
         }
     }
 
     Ok(result?)
 }
 
+fn choose_logs_to_delete(paths: &[PathBuf]) -> Vec<&Path> {
+    let mut most_recent_stem = None;
+    for path in paths {
+        if path.extension() != Some(OsStr::new("log")) {
+            continue;
+        }
+        let stem = path
+            .file_stem()
+            .expect("Every file in the log dir should have a stem");
```

Comment:
I'd rather not make an strong assumption about the path but simply `continue` if this fails.